core/camera/src/main/java/com/google/jetpackcamera/core/camera/CameraXCameraUseCase.kt

```diff
@@ -560,12 +565,24 @@ constructor(
                 .start(callbackExecutor) { onVideoRecordEvent ->
                     run {
                         Log.d(TAG, onVideoRecordEvent.toString())
+                        val isTorchRequired = camera.cameraInfo.hasFlashUnit() &&
+                            currentSettings.value?.flashMode == FlashMode.ON
                         when (onVideoRecordEvent) {
+                            is VideoRecordEvent.Start -> {
+                                if (isTorchRequired) {
+                                    camera.cameraControl.enableTorch(true)
```

Comment:
This should probably happen before we start the recording. Otherwise the beginning of the video may have a larger period where the torch isn't on.

Reply:
Relocating it before the `PendingRecording.start()` call. Additionally, wrap the `start()` method within a try-catch block to handle any exceptions that might arise, ensuring the torch is properly closed.